lib/dialects/mssql/transaction.js

```diff
@@ -55,48 +55,38 @@ module.exports = class Transaction_MSSQL extends Transaction {
   // Acquire a connection and create a disposer - either using the one passed
   // via config or getting one off the client. The disposer will be called once
   // the original promise is marked completed.
-  acquireConnection(config, cb) {
+  async acquireConnection(config, cb) {
     const configConnection = config && config.connection;
-    return new Promise((resolve, reject) => {
-      try {
-        resolve(
-          (this.outerTx ? this.outerTx.conn : null) ||
-            configConnection ||
-            this.client.acquireConnection()
-        );
-      } catch (e) {
-        reject(e);
+    const conn =
+      (this.outerTx && this.outerTx.conn) ||
+      configConnection ||
+      (await this.client.acquireConnection());
+
+    try {
+      conn.__knexTxId = this.txid;
+      if (!this.outerTx) {
```

Comment:
this part was outside try catch

Reply:
Yeah -- there were a handful of lines that were outside of the `try/finally` in the previous code.  But, because the connection had already been established, they really should have been inside the `try` block.  Otherwise, errors would cause connections to be unreleased.

(Granted: the possibility of an error in that code was extremely low.  This is mostly just a precaution)

Reply:
Ok